vcd/resource_vcd_vapp_vm.go

```diff
@@ -904,7 +904,10 @@ func resourceVmHotUpdate(d *schema.ResourceData, meta interface{}, vmType typeOf
 	}
 
 	// due the bug in VCD 10.1 hot update possible for adding new or update existing network, removing of network has to be done with cold update
-	if d.HasChange("network") && !isNetworkRemovedInVcd101(d, meta) {
+	// There are a few cases when hot NIC updates are not possible:
+	// * VCD 10.1 does not allow to remove NICs
+	// * Primary NIC cannot be removed on a powered on VM
```

Comment:
Should this state:
```suggestion
	// * Primary NIC cannot be removed on a powered on VM as of VCD 10.2.2.1 and 10.3
```

Just to confirm, is there any supported VCD version where hot removal still works?

Reply:
Resolving this conversation, as we remove support for hot primary NIC removal for **all** VCD versions.